ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog300.java

```diff
@@ -561,6 +564,32 @@ protected void executeDMLUpdates() throws AmbariException, SQLException {
     updateKerberosConfigurations();
     upgradeLdapConfiguration();
     createRoleAuthorizations();
+    addUserAuthenticationSequence();
+  }
+
+  protected void addUserAuthenticationSequence() throws SQLException {
+    final long maxUserAuthenticationId = fetchMaxUserAuthenticationId();
+    LOG.info("Maximum user authentication ID = " + maxUserAuthenticationId);
+    addSequence("user_authentication_id_seq", maxUserAuthenticationId + 1, false);
+  }
+
+  private long fetchMaxUserAuthenticationId() throws SQLException {
```

Comment:
Would it make sense to create a reusable method that takes the ID column name and table name?

Reply:
Good idea; I'll push a new version

Reply:
+1 for a reusable method.